compositor_render/src/renderer.rs

```diff
@@ -105,15 +111,19 @@ impl Renderer {
             text_renderer_ctx: &self.text_renderer_ctx,
         };
 
+        let scope = WgpuErrorScope::push(&ctx.wgpu_ctx.device);
```

Comment:
this wouldn't work. in this macro, the insides are turned into a closure. the question mark operator would return the error from the closure, which would be ignored, since the closure is called in the `drop` method of the scope guard.